### ISSUE TITLE
Asakuki oil diffuser: fan.asakuki_aroma_diffuser not changing fan as expected

### ISSUE THREAD

**Issue by apuntigam** [make-all/tuya-local#1140]
When using the Asakuki aroma diffuser in Home Assistant via the Tuya Local HACS integration, the fan does not function as expected.

If I go into entities in Home Assistant and play with the settings, I see the following which I think is a bug:

fan.asakuki_aroma_diffuser shows a switch with 3 levels: off, low and high, which is correct for this model where the diffuser has an off, "weak" and "strong" mode.  The problem is, toggling between these on the entity switch does nothing except turn the light on and off - it does not change the strength of the diffuser so it is not acting on the fan. 

If I add the device to my dashboard and turn on the fan, as in the attached screenshot, only the light comes on and the fan does not come on.

![image](https://github.com/make-all/tuya-local/assets/11870335/5aa2c876-1f56-4fe3-b5aa-8b3007fa0763)

I've included the diagnostics.  Also included the log from after the Tuya Local installation.

[tuya_local-bcf19380811bb18c418211a715b0c803-Asakuki aroma diffuser-62a35f153699629c49fcfadfad3b4fa0.json.txt](https://github.com/make-all/tuya-local/files/12804493/tuya_local-bcf19380811bb18c418211a715b0c803-Asakuki.aroma.diffuser-62a35f153699629c49fcfadfad3b4fa0.json.txt)
[home-assistant_2023-10-04T14-14-21.591Z.log](https://github.com/make-all/tuya-local/files/12804494/home-assistant_2023-10-04T14-14-21.591Z.log)

Then just some other feedback to consider:

light.asakuki_aroma_diffuser shows a percentage slider (0% to 100%), a power button and a color selector where I can select color or mood. When I turn this on the fan comes on, which is the same behavior as in the Tuya Smart app. However, it would be nice if I could switch this on without turning on the fan - i.e. light only. But I don't know if this is possible or if it is baked into the diffuser firmware.
The slider adjusts the brightness ONLY in mood 2 and mood 3, which is the same behavior as in the Tuya Smart app (i.e. it is expected). I have explained this because at first I thought this was broken, I had to go back to the Tuya Smart app to verify that the behavior there is the same.  So it's just a bit counter-intuitive.  Not sure if there is a way to disable the brightness slider when in mood 1? On the Tuya Smart app, in the "gradient" light screen (i.e. the equivalent of mood 1), there is no brightness slider so it's a little more obvious there. 

select.asakuki_aroma_diffuser_mood does allow me to change between the following, so no problems, but perhaps a note to briefly explain what each one is would help (e.g. call it "mood 3 (nightlight)"):
mood 1 shifts the lights over the color spectrum
mood 2 remains on the selected color
mood 3 is the night light

select.asakuki_aroma_diffuser_color does allow me to change the color of the light, however:
- If I am using mood 1, the unit continues to cycle through the color spectrum after I select a color (i.e. no long term change). This is expected behavior so no problem here, but might frustrate some users.
- If I am using mood 3 I can't change the color.  That's normal, in the Tuya Smart app I also can't change the color for the night light.  But again it might be frustrating for some users expecting otherwise.

**Comment by make-all** [comment]
As this device was added based on old cloud API docs and local logs only, there is a high possibility that the assumption that the dps are in the same order as documented is false.

To confirm, it would be useful to have the current QueryThingsDataModel API call result for the device, which seems more complete than any of the other sources of API documentaiton.

**Comment by apuntigam** [comment]
> As this device was added based on old cloud API docs and local logs only, there is a high possibility that the assumption that the dps are in the same order as documented is false.
> 
> To confirm, it would be useful to have the current QueryThingsDataModel API call result for the device, which seems more complete than any of the other sources of API documentaiton.

You're right, the ids have changed and there are a few more ids than are referenced in the current asakuki_diffuser.yaml file, which might allow some additional features/entities.  Unfortunately I have no experience with this so doubt I could edit the yaml file to bring in all the ids and create new features...

**Comment by apuntigam** [comment]
I just removed this diffuser from my home assistant and added it again, but this time instead of adding it as an Asakuki I added it as a ditua_dt1522yn_aromadiffuser (which was another one of the options).  All the settings are there now and everything works, so that must be based on the updated API.  Maybe worth just changing the Asakuki file to be the same as the ditua rather than rewriting the Akasuki...?
